Docs for chain rotation (#1452)
iotaledger/wasp: documentation/docs/guide/chains_and_nodes/chain-rotation.md

# Chain Rotation

## Get public key of wallet address

```bash

> wasp-cli chain address

Reviewer: that command is wrong. should be:
```suggestion
wasp-cli address
```

Reviewer: anyway, we explain that any iota/SMR L1 pub key can be used, and whoever owns that key, basically owns the chain. So in a consortium it would probably be wise to use a multisig setup

```

Copy the public key and pass that to the `--gov-contraller` flag on the `wasp-cli chain deploy` command.

## Start chain

Before starting a chain, all nodes need to trust each other. See [Trust setup](./setting-up-a-chain.md#trust-setup) for instructions on chain peering. Deploy a new chain with the command below

```bash
wasp-cli chain deploy --description "Test Chain" --gov-controller ${wallet_public_key} --committee 0,1 --evm-chainid 1076
```

## Add access nodes

Before rotating the chain, you need to some access nodes. To learn more about access nodes see [chain management](./chain-management.md#changing-access-nodes).

```bash
wasp-cli chain change-access-nodes accept ${node_pub_key}
```

## Run DKG

> Run dkg commannd with new committee indices.

Reviewer: ```suggestion
Update your `wasp-cli` config and run the dkg command with the indices for the new committee members.
```

# Chain Rotation

## Get public key of wallet address

> The “Chain Owner” is the only one who can perform administrative tasks. Any iota/SMR L1 pub key can be used, and whoever owns that key, basically owns the chain. In this example we'll use the public key of the wallet address. In a consortium it would probably be wise to use a multisig setup.

Reviewer: ```suggestion
The “Chain Owner” is the only entity that can perform administrative tasks. Any Iota/Shimmer Layer1 public key can be used. Whoever owns that key, basically owns the chain. 
In this example we'll be using the keys auto-generated by `wasp-cli init`.  In a consortium it would be wise to use a multisig setup.
```

Get wallet public key with the command below

```bash
wasp-cli address
```

Copy the public key and pass that to the `--gov-contraller` flag on the `wasp-cli chain deploy` command.

## Start chain

Before starting a chain, all nodes need to trust each other. See [Trust setup](./setting-up-a-chain.md#trust-setup) for instructions on chain peering. Deploy a new chain with the command below

```bash
wasp-cli chain deploy --description "Test Chain" --gov-controller ${wallet_public_key} --committee 0,1 --evm-chainid 1076
```

## Add access nodes

> Depending on your intentions for rotating the chain, you may need to add some access nodes. If all you want to do is remove existing nodes from the chain then skip this step. To learn more about adding access nodes see [chain management](./chain-management.md#changing-access-nodes). Once you have configuted at least one access node, add their public keys with the command below.

Reviewer: ```suggestion
To introduce new committee nodes, its recommended to add them as access nodes while the previous committee is still in place, so that they can sync the chain state.
If all you want to do is remove existing nodes from the chain then skip this step.

To learn more about adding access nodes see [chain management](./chain-management.md#changing-access-nodes). Once you have configured at least one access node, add their public keys with the command below.
```

```bash
wasp-cli chain change-access-nodes accept ${node_pub_key}
```

> Run that command for each public key.

Reviewer: ```suggestion
Run that command for each new access node public key.
Wait for the new access nodes to be fully synced before issuing the `rotate` command.
```
